yolox-burn/src/model/yolox.rs

```diff
@@ -0,0 +1,344 @@
+
+        Ok(model)
+    }
+
+    /// YOLOX-L from [`YOLOX: Exceeding YOLO Series in 2021`](https://arxiv.org/abs/2107.08430).
+    ///
+    /// # Arguments
+    ///
+    /// * `num_classes`: Number of output classes of the model.
+    /// * `device` - Device to create the module on.
+    ///
+    /// # Returns
+    ///
+    /// A YOLOX-L module.
+    pub fn yolox_l(num_classes: usize, device: &Device<B>) -> Self {
+        YoloxConfig::new(1., 1., num_classes, false).init(device)
+    }
+
+    /// YOLOX-L from [`YOLOX: Exceeding YOLO Series in 2021`](https://arxiv.org/abs/2107.08430)
+    /// with pre-trained weights.
+    ///
+    /// # Arguments
+    ///
+    /// * `num_classes`: Number of output classes of the model.
+    /// * `device` - Device to create the module on.
+    ///
+    /// # Returns
+    ///
+    /// A YOLOX-L module with pre-trained weights.
+    #[cfg(feature = "pretrained")]
+    pub fn yolox_l_pretrained(
+        weights: weights::YoloxL,
+        device: &Device<B>,
+    ) -> Result<Self, RecorderError> {
+        let weights = weights.weights();
+        let record = Self::load_weights_record(&weights, device)?;
+
+        let model = YoloxConfig::new(1., 1., weights.num_classes, false).init_with(record);
+
+        Ok(model)
+    }
+
+    /// YOLOX-X from [`YOLOX: Exceeding YOLO Series in 2021`](https://arxiv.org/abs/2107.08430).
+    ///
+    /// # Arguments
+    ///
+    /// * `num_classes`: Number of output classes of the model.
+    /// * `device` - Device to create the module on.
+    ///
+    /// # Returns
+    ///
+    /// A YOLOX-X module.
+    pub fn yolox_x(num_classes: usize, device: &Device<B>) -> Self {
+        YoloxConfig::new(1.33, 1.25, num_classes, false).init(device)
+    }
+
+    /// YOLOX-X from [`YOLOX: Exceeding YOLO Series in 2021`](https://arxiv.org/abs/2107.08430)
+    /// with pre-trained weights.
+    ///
+    /// # Arguments
+    ///
+    /// * `num_classes`: Number of output classes of the model.
+    /// * `device` - Device to create the module on.
+    ///
+    /// # Returns
+    ///
+    /// A YOLOX-X module with pre-trained weights.
+    #[cfg(feature = "pretrained")]
+    pub fn yolox_x_pretrained(
+        weights: weights::YoloxX,
+        device: &Device<B>,
+    ) -> Result<Self, RecorderError> {
+        let weights = weights.weights();
+        let record = Self::load_weights_record(&weights, device)?;
+
+        let model = YoloxConfig::new(1.33, 1.25, weights.num_classes, false).init_with(record);
+
+        Ok(model)
+    }
+
+    /// Load specified pre-trained PyTorch weights as a record.
+    fn load_weights_record(
+        weights: &weights::Weights,
+        device: &Device<B>,
+    ) -> Result<YoloxRecord<B>, RecorderError> {
+        // Download torch weights
+        let torch_weights = weights.download().map_err(|err| {
+            RecorderError::Unknown(format!("Could not download weights.\nError: {err}"))
+        })?;
+
```

Comment:
It would be great also if we resave to Burn native format. PyTorch format is slower and has more dependency. 

Not required to fix this now but we should be thinking it doing it. We can enhance in a next PR.

Reply:
Yeah we could do that for all the imported weights. We just need to store them somewhere, possibly on HF hub? We could keep the discussion going [here](https://github.com/tracel-ai/models/issues/16)